schema.py

```diff
@@ -1,5 +1,36 @@
 __version__ = '0.3.0'
 
+error_messages = {
+    'did not validate': '{schema} {reason} {data}',
```

Comment:
I feel like these should be different subclasses of SchemaError. Then we could catch them selectively with try/except in calling code. What do you think?